smapply/api.py

```diff
@@ -243,6 +249,14 @@ def parse_webhook_user(webhook):
         user = process_user(serializer.data)
     else:
         user = profile.user
+
+    # Sync user demographics data
+    try:
+        store_demographics_data(user.profile, webhook.submission_id)
```

Comment:
Why did you remove submission_id from required fields, but you are expecting it to be present in the webhook here?

Reply:
I didn't. I only removed `award`.

Reply:
`required_fields = field_mapping.keys()`. In this case keys was `id, award, user_id`.

Reply:
It looks like you are setting required fields to this `required_fields = ['id', 'user_id']` on line 220.

Reply:
My intention was to use this function for every webhook, and not all of them will have an `award`. The code that creates the PersonalPrice objects already handles this (`if webhook.award_id is not None:`). So I removed `award` from required fields.

Reply:
In my first comment I was talking about `submission_id`, I did not mention anything about award_id.  Because in your call `store_demographics_data(user.profile, webhook.submission_id)` you are relying on submission_id, so why remove it from required fields?

Reply:
https://github.com/mitodl/bootcamp-ecommerce/pull/300/files#r328303582